src/main/java/frc/robot/poseestimator/limelights/MultiLimelightsRawData.java

```diff
@@ -0,0 +1,43 @@
+package frc.robot.poseestimator.limelights;
+
+import edu.wpi.first.math.Pair;
+import edu.wpi.first.math.geometry.Pose2d;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class MultiLimelightsRawData {
+
+	private final List<Limelight> limelights;
+	private final String logPath;
+
+	public MultiLimelightsRawData(String[] names, String hardwareLogPath) {
+		this.logPath = hardwareLogPath;
+		this.limelights = new ArrayList<>();
+
+		for (String limelightName : names) {
+			limelights.add(new Limelight(limelightName, hardwareLogPath));
+		}
+	}
+
+	public List<LimelightRawData> getAllAvailableLimelightData() {
+		List<LimelightRawData> limelightsData = new ArrayList<>();
+
+		for (Limelight limelight : limelights) {
+			Optional<Pair<Pose2d, Double>> observation = limelight.getUpdatedPose2DEstimation();
+			if (observation.isPresent()) {
+				LimelightRawData limelightRawData = new LimelightRawData(
+					observation.get().getFirst(),
+					limelight.getAprilTagHeight(),
+					limelight.getDistanceFromAprilTag(),
+					observation.get().getSecond()
+				);
+				limelightsData.add(limelightRawData);
+			}
+		}
```

Comment:
maybe pass the apriltag data as input to the getAprilTagHeight() and getDistanceFromAprilTag()? It doesn't make sense that the aprilTag pose is calculated in getUpdated2DEstimation() but it only returns the robot pose

Reply:
I solved it by having a separate function to update its data. I want the data of both arrays to be updated together